src/effects/Echo.cpp

```diff
@@ -85,10 +122,13 @@ unsigned EffectEcho::GetAudioOutCount() const
    return 1;
 }
 
-bool EffectEcho::ProcessInitialize(
-   EffectSettings &, sampleCount, ChannelNames)
+bool EffectEcho::Instance::ProcessInitialize(
+   EffectSettings& settings, sampleCount, ChannelNames)
```

Comment:
Do NOT name and the settings argument in step 3.  Add that only at step 4.

And step 3, static_cast mProcessor to EffectEcho and use its mSettings.  At step 4, remove those lines, as you must to compile without the mSettings member.  Only then use the arguments.

Reply:
ok, done